test/unit/reductions/grid_search/test_grid_search_demographicparity.py

```diff
@@ -222,3 +223,36 @@ def test_can_specify_and_generate_lambda_vecs(A_two_dim):
         coef1 = grid_search1._predictors[i].coef_
         coef2 = grid_search2._predictors[i].coef_
         assert np.array_equal(coef1, coef2)
+
+
+def test_constant_predictor():
+    # Setup with data designed to result in "all single class"
+    # at some point in the grid
+    X_dict = {
+        "c": [0, 1, 4, 1, 5, 1, 6, 0, 2, 4],
+        "d": [1, 5, 1, 6, 2, 3, 5, 1, 5, 2]
+    }
+    X = pd.DataFrame(X_dict)
+
+    y = [0, 0, 0, 0, 0, 0, 0, 0, 0, 1]
+    A = [3, 3, 3, 3, 3, 3, 3, 3, 3, 11]
```

Comment:
Any specific reason for 3 and 11? Might confuse people. Perhaps best to make it "A" and "B" or at least something simpler than random numbers. Or comment that these are just group identifiers

Reply:
Will do. Quotes are more tedious to type.